integration/main_test.go

```diff
@@ -805,9 +805,11 @@ var _ = Describe("Router Integration", func() {
 				req, err := http.NewRequest("GET", fmt.Sprintf("http://%s:%d", localIP, proxyPort), nil)
 				Expect(err).ToNot(HaveOccurred())
 				req.Host = "demo." + test_util.LocalhostDNS
-				res, err := client.Do(req)
-				Expect(err).ToNot(HaveOccurred())
-				Expect(res.StatusCode).To(Equal(TEST_STATUS_CODE))
+				Eventually(func() int {
```

Comment:
For me, the integration-suite tests have different issues locally before and after this change.
Maybe it depends on the tester-system, keeping it in this PR as it can only improve the current status from my POV

Reply:
Before I had consistent failures on this particular test, now it's rare.

Reply:
True, same for me, thx for this enhancement!